### PR TITLE
Add practice exercise `rational-numbers`

### PR DESCRIPTION
**Pull request by natanaelsirqueira**

### CODE REVIEW

File: exercises/practice/rational-numbers/.meta/example.gleam [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      RationalNumber(
+        numerator: int.negate(numerator),
+        denominator: int.negate(denominator),
+      )
+
+    numerator, denominator -> RationalNumber(numerator, denominator)
+  }
+}
+
+fn power_of_integer(base: Int, to exponent: Int) -> Result(Int, Nil) {
```
Comment (jiegillet) [inline exercises/practice/rational-numbers/.meta/example.gleam:94]: It's a bit strange to have labels on some arguments but not others, I could have sworn the compiler would complain...

File: exercises/practice/rational-numbers/.meta/example.gleam [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+      Ok(power)
+    }
+  }
+}
+
+pub fn power_of_real(
+  number base: Int,
+  to exponent: RationalNumber,
+) -> Result(Float, Nil) {
```
Comment (jiegillet) [inline exercises/practice/rational-numbers/.meta/example.gleam:70]: It feels strange to have a `Result` here but no test with an error case...
What do you think? 
Maybe we should add a test case with a negative base to the problem-specs?

File: exercises/practice/rational-numbers/.meta/example.gleam [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        to: int.absolute_value(exponent),
+      )
+
+    RationalNumber(numerator, denominator) -> {
+      try numerator = power_of_integer(numerator, to: exponent)
+      try denominator = power_of_integer(denominator, to: exponent)
+
+      let power = reduce(RationalNumber(numerator, denominator))
+
+      Ok(power)
```
Comment (jiegillet) [inline exercises/practice/rational-numbers/.meta/example.gleam:62]: I don't think a rational to an integer power can actually fail, can it?

File: exercises/practice/rational-numbers/.meta/example.gleam [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+  reduce(RationalNumber(numerator: a * d, denominator: c * b))
+}
+
+pub fn absolute_value(r: RationalNumber) -> RationalNumber {
+  let RationalNumber(numerator, denominator) = reduce(r)
+
+  RationalNumber(
+    numerator: int.absolute_value(numerator),
+    denominator: int.absolute_value(denominator),
```
Comment (jiegillet) [inline exercises/practice/rational-numbers/.meta/example.gleam:41]: Isn't the denominator always positive after reducing?

**Review by jiegillet — COMMENTED** [review]
Nice work! Let's iron out some wrinkles :)

File: exercises/practice/rational-numbers/.meta/example.gleam (hunk 3 above)
Comment (natanaelsirqueira) [reply exercises/practice/rational-numbers/.meta/example.gleam:62]: So... I did this because `power_of_integer` can fail (because `int.power` can fail). Should we make it so it doesn't return a `Result`?

```gleam
fn power_of_integer(number base: Int, to exponent: Int) -> Int {
  assert Ok(power) = int.power(base, int.to_float(exponent))

  float.round(power)
}
```

We could also just use `assert` on lines 57 and 58 instead:

```gleam
    RationalNumber(numerator, denominator) -> {
      assert Ok(numerator) = power_of_integer(numerator, to: exponent)
      assert Ok(denominator) = power_of_integer(denominator, to: exponent)

      reduce(RationalNumber(numerator, denominator))
    }
```

File: exercises/practice/rational-numbers/.meta/example.gleam (hunk 2 above)
Comment (natanaelsirqueira) [reply exercises/practice/rational-numbers/.meta/example.gleam:70]: Sounds good. I'll work on a PR later today.

Comment (natanaelsirqueira) [reply exercises/practice/rational-numbers/.meta/example.gleam:70]: Opened the [PR](https://github.com/exercism/problem-specifications/pull/2213) and it got closed as expected. Are you able to reopen?

Comment (jiegillet) [reply exercises/practice/rational-numbers/.meta/example.gleam:70]: Thank you. Let's wait for the PR to be merged before merging this one :)

File: exercises/practice/rational-numbers/.meta/example.gleam (hunk 3 above)
Comment (jiegillet) [reply exercises/practice/rational-numbers/.meta/example.gleam:62]: `int.power` can fail when the exponent is fractional, but we know that it is integer and positive here, so we can safely use `assert`. We even skip `power_of_integer` entirely and use `int.power`, but that's a matter of style.
